crates/libs/result/natvis/windows-result.natvis

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+  <Type Name="windows_result::error::Error">
+    <Expand>
+      <ExpandedItem>code</ExpandedItem>
+      <Item Name="[info]">info</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="windows_result::hresult::HRESULT">
+    <DisplayString>{(HRESULT)__0}</DisplayString>
+  </Type>
```

Comment:
Don't think this HRESULT visualizer works properly in some cases. For example, `let z = windows_result::Error::from_win32();` results in:

```
Error: no suitable constructor exists to convert from "int" to "windows_result::hresult::HRESULT"
```

![image](https://github.com/microsoft/windows-rs/assets/475132/5fccbbab-d064-4d54-86dc-1262891d5fc2)

Reply:
This looks just about right. `Error::from_win32()` is the equivalent of adding an `@err,hr` debug [watch](https://learn.microsoft.com/en-us/visualstudio/debugger/watch-and-quickwatch-windows). The reported value never makes sense until it does. The specific value `-2147024854` is `0x8007002A` (in hex), i.e., a [Win32 error code](https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-) with value 42. That value isn't assigned to a particular error condition (curiously).

That doesn't explain, though, why the natvis EE subscribes this specific `i32` to an `Option`. I guess I saw this coming all along, and somehow managed to keep my name out of the [`git blame` game](https://devblogs.microsoft.com/oldnewthing/20190916-00/?p=102892).

To address this we would need to understand what's going on inside the natvis EE. I'd `@`-ping wesleywiser, again, but they seem to be having a busy week 😉.

Reply:
The screenshot was just to visually show the natvis expression failed; the 42 value was just my test value.

Here's a workaround that should cover all our bases:

```suggestion
pe Name="windows_result::hresult::HRESULT">
    <DisplayString Optional="true">{(HRESULT)__0,hr}</DisplayString> <!-- cdb, windbg, ... -->
    <DisplayString Optional="true">{__0,hr}</DisplayString> <!-- Visual Studio -->
</Type>
```

Reply:
@riverar I can reproduce the issue with Visual Studio (CDB, WinDbg Preview and WinDbg all work as expected). However, the proposed change doesn't have any effect for me:

![image](https://github.com/microsoft/windows-rs/assets/23303506/92f860e0-5692-4624-9a87-b7f45d49cb9f)

Notably, there is no `[info]` expansion for me either. It seems that the `windows_result::error::Error` rule isn't applied, so maybe I'm seeing a side effect of the first rule failing.

What do you (and Kenny) propose going forward? Merge Rafael's suggested change? Leaving this PR open for now and move ahead with the 0.53 release?

Reply:
Ah right. Let's just forget about Visual Studio visualization support for now (i.e. ignore my suggestion above) and we'll follow up with a test suite and new natvis files.